pontoon/checks/libraries/pontoon.py

```diff
@@ -37,36 +37,39 @@ def run_checks(entity, string):
     checks = defaultdict(list)
     resource_ext = entity.resource.format
 
-    # Prevent translations exceeding the given length limit
-    max_length = get_max_length(entity.comment)
-    html_parser = HTMLParser.HTMLParser()
-    string_length = len(
-        html_parser.unescape(
-            bleach.clean(
-                string,
-                strip=True,
-                tags=()
+    if resource_ext == 'lang':
+        # Newlines are not allowed in .lang files (bug 1190754)
+        if '\n' in string:
+            checks['pErrors'].append(
+                'Newline characters are not allowed'
             )
-        )
-    )
 
-    if max_length and string_length > max_length:
-        checks['pErrors'].append(
-            'Translation too long'
-        )
+        # Prevent translations exceeding the given length limit
+        max_length = get_max_length(entity.comment)
+
+        if max_length:
+            html_parser = HTMLParser.HTMLParser()
```

Comment:
Move this line to the top of the file, after `parser = FluentParser()` and we're good to merge this.